6-task-queueing/lib/active_job/queue_adapters/pub_sub_queue_adapter.rb

```diff
@@ -12,35 +12,49 @@ def self.pubsub
         end
       end
 
+      def self.pubsub_topic
+        @pubsub_topic ||= Rails.application.config.x.settings["pubsub_topic"]
+      end
+
+      def self.pubsub_subscription
+        @pubsub_subscription ||= Rails.application.config.x.settings["pubsub_subscription"]
+      end
+
       def self.enqueue job
         Rails.logger.info "[PubSubQueueAdapter] enqueue job #{job.inspect}"
 
         book  = job.arguments.first
-        topic = pubsub.topic "lookup_book_details_queue"
+
+        topic = pubsub.topic pubsub_topic
 
         topic.publish book.id.to_s
       end
 # [END pub_sub_enqueue]
 
-      # TODO add queue parameter
-
       # [START pub_sub_worker]
       def self.run_worker!
         Rails.logger.info "Running worker to lookup book details"
 
-        topic        = pubsub.topic       "lookup_book_details_queue"
-        subscription = topic.subscription "lookup_book_details"
+        topic        = pubsub.topic pubsub_topic
+        subscription = topic.subscription pubsub_subscription
 
-        topic.subscribe "lookup_book_details" unless subscription.exists?
+        subscriber = subscription.listen do |message|
+          message.acknowledge!
 
-        subscription.listen autoack: true do |message|
           Rails.logger.info "Book lookup request (#{message.data})"
 
           book_id = message.data.to_i
           book    = Book.find_by_id book_id
 
           LookupBookDetailsJob.perform_now book if book
         end
+
+        # Start background threads that will call block passed to listen.
+        subscriber.start
+
+        loop do
+          sleep 10
+        end
```

Comment:
This is kind of weird. If you want to sleep indefinitely, just call `sleep` without an argument.

Reply:
:man_facepalming:  thanks!